src/treelearner/cuda/cuda_single_gpu_tree_learner.cpp

```diff
@@ -562,9 +570,7 @@ void CUDASingleGPUTreeLearner::SelectFeatureByNode(const Tree* tree) {
 #ifdef DEBUG
 void CUDASingleGPUTreeLearner::CheckSplitValid(
   const int left_leaf,
-  const int right_leaf,
-  const double split_sum_left_gradients,
-  const double split_sum_right_gradients) {
+  const int right_leaf) {
```

Comment:
Could you help me understand this change? What was wrong with the prior approach?

I'm just unsure how this change is related to the goal "fix `dump_model()` information for root node".

Reply:
It simplifies the code, introducing the `leaf_sum_gradients` which is needed for calculating the sum of the gradients in the root before the training starts. This was the most straightforward way to do it in my eyes.